bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -307,6 +309,14 @@ public String getLabel(Widget w) {
                             }
                         } else {
                             state = item.getState();
+                            if (stateDescription != null && stateDescription.getOptions() != null) {
```

Comment:
It is the fix for options. If you have options defined for a string channel, geLabel(w) should translate the value to its label. By the way, I think my code is not ewactly at the right place, I have to consider the case where the user requests for a transformation.